// @ai-sdk/togetherai@2.0.2/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  VERSION: () => VERSION,
  createTogetherAI: () => createTogetherAI,
  togetherai: () => togetherai
});
module.exports = __toCommonJS(src_exports);

// src/togetherai-provider.ts
var import_openai_compatible = require("@ai-sdk/openai-compatible");
var import_provider_utils5 = require("@ai-sdk/provider-utils");

// src/reranking/togetherai-reranking-model.ts
var import_provider_utils3 = require("@ai-sdk/provider-utils");

// src/reranking/togetherai-reranking-api.ts
var import_provider_utils = require("@ai-sdk/provider-utils");
var import_v4 = require("zod/v4");
var togetheraiErrorSchema = (0, import_provider_utils.lazySchema)(
  () => (0, import_provider_utils.zodSchema)(
    import_v4.z.object({
      error: import_v4.z.object({
        message: import_v4.z.string()
      })
    })
  )
);
var togetheraiRerankingResponseSchema = (0, import_provider_utils.lazySchema)(
  () => (0, import_provider_utils.zodSchema)(
    import_v4.z.object({
      id: import_v4.z.string().nullish(),
      model: import_v4.z.string().nullish(),
      results: import_v4.z.array(
        import_v4.z.object({
          index: import_v4.z.number(),
          relevance_score: import_v4.z.number()
        })
      ),
      usage: import_v4.z.object({
        prompt_tokens: import_v4.z.number(),
        completion_tokens: import_v4.z.number(),
        total_tokens: import_v4.z.number()
      })
    })
  )
);

// src/reranking/togetherai-reranking-options.ts
var import_provider_utils2 = require("@ai-sdk/provider-utils");
var import_v42 = require("zod/v4");
var togetheraiRerankingOptionsSchema = (0, import_provider_utils2.lazySchema)(
  () => (0, import_provider_utils2.zodSchema)(
    import_v42.z.object({
      rankFields: import_v42.z.array(import_v42.z.string()).optional()
    })
  )
);

// src/reranking/togetherai-reranking-model.ts
var TogetherAIRerankingModel = class {
  constructor(modelId, config) {
    this.specificationVersion = "v3";
    this.modelId = modelId;
    this.config = config;
  }
  get provider() {
    return this.config.provider;
  }
  // see https://docs.together.ai/reference/rerank-1
  async doRerank({
    documents,
    headers,
    query,
    topN,
    abortSignal,
    providerOptions
  }) {
    var _a, _b;
    const rerankingOptions = await (0, import_provider_utils3.parseProviderOptions)({
      provider: "togetherai",
      providerOptions,
      schema: togetheraiRerankingOptionsSchema
    });
    const {
      responseHeaders,
      value: response,
      rawValue
    } = await (0, import_provider_utils3.postJsonToApi)({
      url: `${this.config.baseURL}/rerank`,
      headers: (0, import_provider_utils3.combineHeaders)(this.config.headers(), headers),
      body: {
        model: this.modelId,
        documents: documents.values,
        query,
        top_n: topN,
        rank_fields: rerankingOptions == null ? void 0 : rerankingOptions.rankFields,
        return_documents: false
        // reduce response size
      },
      failedResponseHandler: (0, import_provider_utils3.createJsonErrorResponseHandler)({
        errorSchema: togetheraiErrorSchema,
        errorToMessage: (data) => data.error.message
      }),
      successfulResponseHandler: (0, import_provider_utils3.createJsonResponseHandler)(
        togetheraiRerankingResponseSchema
      ),
      abortSignal,
      fetch: this.config.fetch
    });
    return {
      ranking: response.results.map((result) => ({
        index: result.index,
        relevanceScore: result.relevance_score
      })),
      response: {
        id: (_a = response.id) != null ? _a : void 0,
        modelId: (_b = response.model) != null ? _b : void 0,
        headers: responseHeaders,
        body: rawValue
      }
    };
  }
};

// src/togetherai-image-model.ts
var import_provider_utils4 = require("@ai-sdk/provider-utils");
var import_v43 = require("zod/v4");
var TogetherAIImageModel = class {
  constructor(modelId, config) {
    this.modelId = modelId;
    this.config = config;
    this.specificationVersion = "v3";
    this.maxImagesPerCall = 1;
  }
  get provider() {
    return this.config.provider;
  }
  async doGenerate({
    prompt,
    n,
    size,
    seed,
    providerOptions,
    headers,
    abortSignal,
    files,
    mask
  }) {
    var _a, _b, _c;
    const warnings = [];
    if (mask != null) {
      throw new Error(
        "Together AI does not support mask-based image editing. Use FLUX Kontext models (e.g., black-forest-labs/FLUX.1-kontext-pro) with a reference image and descriptive prompt instead."
      );
    }
    if (size != null) {
      warnings.push({
        type: "unsupported",
        feature: "aspectRatio",
        details: "This model does not support the `aspectRatio` option. Use `size` instead."
      });
    }
    const currentDate = (_c = (_b = (_a = this.config._internal) == null ? void 0 : _a.currentDate) == null ? void 0 : _b.call(_a)) != null ? _c : /* @__PURE__ */ new Date();
    const togetheraiOptions = await (0, import_provider_utils4.parseProviderOptions)({
      provider: "togetherai",
      providerOptions,
      schema: togetheraiImageProviderOptionsSchema
    });
    let imageUrl;
    if (files != null && files.length > 0) {
      imageUrl = (0, import_provider_utils4.convertImageModelFileToDataUri)(files[0]);
      if (files.length > 1) {
        warnings.push({
          type: "other",
          message: "Together AI only supports a single input image. Additional images are ignored."
        });
      }
    }
    const splitSize = size == null ? void 0 : size.split("x");
    const { value: response, responseHeaders } = await (0, import_provider_utils4.postJsonToApi)({
      url: `${this.config.baseURL}/images/generations`,
      headers: (0, import_provider_utils4.combineHeaders)(this.config.headers(), headers),
      body: {
        model: this.modelId,
        prompt,
        seed,
        n,
        ...splitSize && {
          width: parseInt(splitSize[0]),
          height: parseInt(splitSize[1])
        },
        ...imageUrl != null ? { image_url: imageUrl } : {},
        response_format: "base64",
        ...togetheraiOptions != null ? togetheraiOptions : {}
      },
      failedResponseHandler: (0, import_provider_utils4.createJsonErrorResponseHandler)({
        errorSchema: togetheraiErrorSchema2,
        errorToMessage: (data) => data.error.message
      }),
      successfulResponseHandler: (0, import_provider_utils4.createJsonResponseHandler)(
        togetheraiImageResponseSchema
      ),
      abortSignal,
      fetch: this.config.fetch
    });
    return {
      images: response.data.map((item) => item.b64_json),
      warnings,
      response: {
        timestamp: currentDate,
        modelId: this.modelId,
        headers: responseHeaders
      }
    };
  }
};
var togetheraiImageResponseSchema = import_v43.z.object({
  data: import_v43.z.array(
    import_v43.z.object({
      b64_json: import_v43.z.string()
    })
  )
});
var togetheraiErrorSchema2 = import_v43.z.object({
  error: import_v43.z.object({
    message: import_v43.z.string()
  })
});
var togetheraiImageProviderOptionsSchema = (0, import_provider_utils4.lazySchema)(
  () => (0, import_provider_utils4.zodSchema)(
    import_v43.z.object({
      /**
       * Number of generation steps. Higher values can improve quality.
       */
      steps: import_v43.z.number().nullish(),
      /**
       * Guidance scale for image generation.
       */
      guidance: import_v43.z.number().nullish(),
      /**
       * Negative prompt to guide what to avoid.
       */
      negative_prompt: import_v43.z.string().nullish(),
      /**
       * Disable the safety checker for image generation.
       * When true, the API will not reject images flagged as potentially NSFW.
       * Not available for Flux Schnell Free and Flux Pro models.
       */
      disable_safety_checker: import_v43.z.boolean().nullish()
    }).passthrough()
  )
);

// src/version.ts
var VERSION = true ? "2.0.2" : "0.0.0-test";

// src/togetherai-provider.ts
function createTogetherAI(options = {}) {
  var _a;
  const baseURL = (0, import_provider_utils5.withoutTrailingSlash)(
    (_a = options.baseURL) != null ? _a : "https://api.together.xyz/v1/"
  );
  const getHeaders = () => (0, import_provider_utils5.withUserAgentSuffix)(
    {
      Authorization: `Bearer ${(0, import_provider_utils5.loadApiKey)({
        apiKey: options.apiKey,
        environmentVariableName: "TOGETHER_AI_API_KEY",
        description: "TogetherAI"
      })}`,
      ...options.headers
    },
    `ai-sdk/togetherai/${VERSION}`
  );
  const getCommonModelConfig = (modelType) => ({
    provider: `togetherai.${modelType}`,
    url: ({ path }) => `${baseURL}${path}`,
    headers: getHeaders,
    fetch: options.fetch
  });
  const createChatModel = (modelId) => {
    return new import_openai_compatible.OpenAICompatibleChatLanguageModel(
      modelId,
      getCommonModelConfig("chat")
    );
  };
  const createCompletionModel = (modelId) => new import_openai_compatible.OpenAICompatibleCompletionLanguageModel(
    modelId,
    getCommonModelConfig("completion")
  );
  const createEmbeddingModel = (modelId) => new import_openai_compatible.OpenAICompatibleEmbeddingModel(
    modelId,
    getCommonModelConfig("embedding")
  );
  const createImageModel = (modelId) => new TogetherAIImageModel(modelId, {
    ...getCommonModelConfig("image"),
    baseURL: baseURL != null ? baseURL : "https://api.together.xyz/v1/"
  });
  const createRerankingModel = (modelId) => new TogetherAIRerankingModel(modelId, {
    ...getCommonModelConfig("reranking"),
    baseURL: baseURL != null ? baseURL : "https://api.together.xyz/v1/"
  });
  const provider = (modelId) => createChatModel(modelId);
  provider.specificationVersion = "v3";
  provider.completionModel = createCompletionModel;
  provider.languageModel = createChatModel;
  provider.chatModel = createChatModel;
  provider.embeddingModel = createEmbeddingModel;
  provider.textEmbeddingModel = createEmbeddingModel;
  provider.image = createImageModel;
  provider.imageModel = createImageModel;
  provider.reranking = createRerankingModel;
  provider.rerankingModel = createRerankingModel;
  return provider;
}
var togetherai = createTogetherAI();
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  VERSION,
  createTogetherAI,
  togetherai
});
